tests/test_files.py

```diff
@@ -0,0 +1,182 @@
+    assert gh.getitem_url == 'https://api.github.com/repos/cpython/python/issue/1234'
+    assert len(gh.post_url) == 2
+    assert gh.post_url[0] == 'https://api.github.com/some/status'
+    assert gh.post_data[0]['state'] == 'failure'
+    assert gh.post_url[1] == 'https://api.github.com/some/label'
+    assert gh.post_data[1] == ['type-documentation']
+
+
+async def test_tests_only():
+    filenames = [{'filename': '/path/to/test_docs1.py'}, {'filename': 'test_docs2.py'}]
+    issue = {'labels': [],
+             'labels_url': 'https://api.github.com/some/label'}
+    gh = FakeGH(getiter=filenames, getitem=issue)
+    event_data = {
+        'action': 'opened',
+        'number': 1234,
+        'pull_request': {
+            'url': 'https://api.github.com/repos/cpython/python/pulls/1234',
+            'statuses_url': 'https://api.github.com/some/status',
+            'issue_url': 'https://api.github.com/repos/cpython/python/issue/1234',
+        },
+    }
+    event = sansio.Event(event_data, event='pull_request', delivery_id=1)
+    await files.router.dispatch(event, gh)
+    assert gh.getiter_url == 'https://api.github.com/repos/cpython/python/pulls/1234/files'
+    assert gh.getitem_url == 'https://api.github.com/repos/cpython/python/issue/1234'
+    assert len(gh.post_url) == 2
+    assert gh.post_url[0] == 'https://api.github.com/some/status'
+    assert gh.post_data[0]['state'] == 'failure'
+    assert gh.post_url[1] == 'https://api.github.com/some/label'
+    assert gh.post_data[1] == ['type-tests']
+
+
+async def test_docs_and_tests():
+    filenames = [{'filename': '/path/to/docs.rst'}, {'filename': 'test_docs2.py'}]
+    issue = {'labels': [{'name': 'skip news'}],
+             'labels_url': 'https://api.github.com/some/label'}
+    gh = FakeGH(getiter=filenames, getitem=issue)
+    event_data = {
+        'action': 'opened',
+        'number': 1234,
+        'pull_request': {
+            'url': 'https://api.github.com/repos/cpython/python/pulls/1234',
+            'statuses_url': 'https://api.github.com/some/status',
+            'issue_url': 'https://api.github.com/repos/cpython/python/issue/1234',
+        },
+    }
+    event = sansio.Event(event_data, event='pull_request', delivery_id=1)
+    await files.router.dispatch(event, gh)
+    assert gh.getiter_url == 'https://api.github.com/repos/cpython/python/pulls/1234/files'
+    assert gh.getitem_url == 'https://api.github.com/repos/cpython/python/issue/1234'
+    # Only creates type-tests label.
+    assert len(gh.post_url) == 2
+    assert gh.post_url[0] == 'https://api.github.com/some/status'
+    assert gh.post_data[0]['state'] == 'success'
+    assert gh.post_url[1] == 'https://api.github.com/some/label'
+    assert gh.post_data[1] == ['type-tests']
+
+
+async def test_news_and_tests():
+    filenames = [{'filename': '/path/to/docs.rst'}, {'filename': 'test_docs2.py'},
+                 {'filename': 'Misc/NEWS.d/next/Lib/' + GOOD_BASENAME}]
```

Comment:
f-string 😄 (and anywhere else it makes sense.)

Reply:
Of course!  I had copied this from the other tests and didn't even think about it.  :-)